js/utils/object.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * @param {any} obj
+ * @returns {boolean}
+ */
+export function is_object(obj) {
+    return obj && obj instanceof Object;
+}
+
+/**
+ * @param {any} to
+ * @param {any} obj
+ * @returns {any}
+ */
+export function deep_assign(to, obj) {
+	if (!is_object(obj) || !is_object(to)) return to;
+	for (const key of Object.keys(obj)) {
```

Comment:
I'm not a fan of this because it treats object-like things as proper objects, e.g. arrays:

```js
console.log([] instanceof Object); // true
console.log(Object.keys(['hi'])); // ['0']
```

That does _work_, but it feels icky.